src/main/java/ui/windows/CreationDisplay.java

```diff
@@ -0,0 +1,147 @@
+        textPanel.add(makeCPUorNo);
+        fieldsAndBoxes.add(textPanel);
+
+        for(int i = 0; i < 6; i++) {
+            JTextField textField = this.textFieldList.get(i);
+            JCheckBox checkBox = this.checkBoxList.get(i);
+            JPanel fieldAndBox = new JPanel();
+            fieldAndBox.setLayout(new BoxLayout(fieldAndBox, BoxLayout.X_AXIS));
+            fieldAndBox.add(textField);
+            fieldAndBox.add(checkBox);
+            this.fieldsAndBoxes.add(fieldAndBox);
+        }
+
+    }
+
+    /**
+     * Create the header message and text-fields for each player name and add the text-fields to the list of text-fields
+     */
+    private void initializeTextFields() {
+        this.giveNameMessage = new JLabel("Give the names of the players.");
+
+        JTextField p1Name = new JTextField();
+        this.textFieldList.add(p1Name);
+        JTextField p2Name = new JTextField();
+        this.textFieldList.add(p2Name);
+        JTextField p3Name = new JTextField();
+        this.textFieldList.add(p3Name);
+        JTextField p4Name = new JTextField();
+        this.textFieldList.add(p4Name);
+        JTextField p5Name = new JTextField();
+        this.textFieldList.add(p5Name);
+        JTextField p6Name = new JTextField();
+        this.textFieldList.add(p6Name);
+    }
+
+    /**
+     * Create all the checkboxes and add them to the list of checkboxes.
+     */
+    private void initializeCheckBoxes() {
+        this.makeCPUorNo = new JLabel("     Check to make player computer-controlled.");
+
+        JCheckBox p1isComp = new JCheckBox();
+        this.checkBoxList.add(p1isComp);
+        JCheckBox p2isComp = new JCheckBox();
+        this.checkBoxList.add(p2isComp);
+        JCheckBox p3isComp = new JCheckBox();
+        this.checkBoxList.add(p3isComp);
+        JCheckBox p4isComp = new JCheckBox();
+        this.checkBoxList.add(p4isComp);
+        JCheckBox p5isComp = new JCheckBox();
+        this.checkBoxList.add(p5isComp);
+        JCheckBox p6isComp = new JCheckBox();
+        this.checkBoxList.add(p6isComp);
+
+    }
+
+    /**
+     * Invoked when an action occurs.
+     * @param e the event to be processed
+     */
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        ArrayList<PlayerCreationInformation> controllerInput = new ArrayList<>();
+        // Take all the names that were entered, and add them alongside the option of whether to make the player
+        // computer-controlled or not.
+        for(int i = 0; i < 6; i ++) {
+            String playerName = textFieldList.get(i).getText();
+            boolean isComputerPlayer = checkBoxList.get(i).isSelected();
+            // Make sure given name is not empty, otherwise, don't add it.
+            if(playerName.length() > 0) {
+                PlayerCreationInformation playerInfo = new PlayerCreationInformation(playerName, isComputerPlayer);
+                controllerInput.add(playerInfo);
```

Comment:
I don't know if it should be this method or not, but should we make sure that no two users can share the same name? By throwing an exception, or something like that.